File: test-rigs/python-test-rig/taskmanager/formatter.py
"""Text formatting utilities for task display."""

from .model import Priority, Task
from .usecase import TaskStats


def format_task_line(task: Task) -> str:
    """Format a single task as a one-line string."""
    status = "[x]" if task.completed else "[ ]"
    priority_label = task.priority.value.upper()
    tags_str = ""
    if task.tags:
        tags_str = " " + " ".join(f"#{t}" for t in task.tags)
    return f"{status} [{priority_label}] {task.title}{tags_str}"


def format_task_list(tasks: list[Task]) -> str:
    """Format a list of tasks as a multi-line string."""
    if not tasks:
        return "No tasks."
    lines = [format_task_line(t) for t in tasks]
    return "\n".join(lines)


def format_stats(stats: TaskStats) -> str:
    """Format task statistics as a summary string."""
    if stats.total == 0:
        return "No tasks tracked."
    pct = (stats.completed / stats.total) * 100
    return (
        f"Tasks: {stats.total} total, {stats.completed} completed, "
        f"{stats.pending} pending ({pct:.1f}% done)"
    )


def format_priority_distribution(stats: TaskStats) -> str:
    """Format priority distribution as a multi-line string."""
    if stats.total == 0:
        return "No tasks to analyze."
    lines = []
    for priority in Priority:
        count = stats.by_priority.get(priority, 0)
        pct = (count / stats.total) * 100
        bar = "#" * int(pct / 5)
        lines.append(f"  {priority.value:>8}: {bar} {count} ({pct:.0f}%)")
    return "\n".join(lines)

File: test-rigs/go-test-rig/formatter/formatter.go
package formatter

import (
	"fmt"
	"strings"

	"github.com/jkjamies/omnivore/go-test-rig/model"
	"github.com/jkjamies/omnivore/go-test-rig/usecase"
)

// FormatTaskLine returns a single-line summary of a task.
func FormatTaskLine(task *model.Task) string {
	status := "○"
	if task.Completed {
		status = "✓"
	}

	var badge string
	switch task.Priority {
	case model.PriorityCritical:
		badge = "[!!!]"
	case model.PriorityHigh:
		badge = "[!!]"
	case model.PriorityMedium:
		badge = "[!]"
	default:
		badge = "[ ]"
	}

	tags := ""
	if len(task.Tags) > 0 {
		tags = fmt.Sprintf(" (%s)", strings.Join(task.Tags, ", "))
	}

	return fmt.Sprintf("%s %s %s%s", status, badge, task.Title, tags)
}

// FormatTaskList formats a slice of tasks as a multi-line report.
func FormatTaskList(tasks []*model.Task) string {
	if len(tasks) == 0 {
		return "No tasks found."
	}
	lines := make([]string, len(tasks))
	for i, t := range tasks {
		lines[i] = FormatTaskLine(t)
	}
	return strings.Join(lines, "\n")
}

// FormatStats formats task statistics as a summary string.
func FormatStats(stats usecase.TaskStats) string {
	return fmt.Sprintf("Tasks: %d total, %d completed, %d pending (%.0f%% done)",
		stats.Total, stats.Completed, stats.Pending, stats.CompletionRate*100)
}

// FormatPriorityDistribution formats priority groups as a table.
func FormatPriorityDistribution(groups []usecase.PriorityGroup) string {
	if len(groups) == 0 {
		return "No tasks."
	}
	lines := make([]string, len(groups))
	for i, g := range groups {
		lines[i] = fmt.Sprintf("%s: %d task(s)", g.Priority.String(), len(g.Tasks))
	}
	return strings.Join(lines, "\n")
}
